snooty/main.py

```diff
@@ -190,32 +195,23 @@ def on_update(
         if page.query_fields:
             document.update({"query_fields": page.query_fields})
 
-        self.client[self.db][COLL_DOCUMENTS].replace_one(
-            document_filter, document, upsert=True
+        self.pending_writes.append(
+            pymongo.ReplaceOne(document_filter, document, upsert=True)
         )
 
-        remote_assets = set(
-            doc["_id"]
-            for doc in self.client[self.db][COLL_ASSETS].find(
-                {"_id": {"$in": checksums}},
-                {"_id": True},
-                cursor_type=pymongo.cursor.CursorType.EXHAUST,
-            )
-        )
-        missing_assets = page.static_assets.difference(remote_assets)
-
-        for static_asset in missing_assets:
-            if not static_asset.can_upload():
-                continue
-
-            self.client[self.db][COLL_ASSETS].replace_one(
-                {"_id": static_asset.get_checksum()},
-                {
-                    "_id": static_asset.get_checksum(),
-                    "filename": str(static_asset.fileid),
-                    "data": static_asset.data,
-                },
-                upsert=True,
+        for static_asset in page.static_assets:
+            self.pending_writes.append(
```

Comment:
nice optimization.

Reply:
To be fair, if we were running this over a dialup connection, it would likely be significantly slower than the previous code! :D